testing/test-setup-config/src/lib/vitest-config-factory.unit.test.ts

```diff
@@ -0,0 +1,356 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { defineConfig } from 'vitest/config';
+import type { E2ETestOptions, TestKind } from './vitest-config-factory.js';
+import { createVitestConfig } from './vitest-config-factory.js';
+
+vi.mock('vitest/config', async importOriginal => {
+  const actual = await importOriginal<typeof import('vitest/config')>();
+  return {
+    ...actual,
+    defineConfig: vi.fn(config => config),
+  };
+});
+
+vi.mock('./vitest-tsconfig-path-aliases.js', () => ({
+  tsconfigPathAliases: vi
+    .fn()
+    .mockReturnValue([{ find: '@test/alias', replacement: '/mock/path' }]),
+}));
+
+describe('createVitestConfig', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('unit test configuration', () => {
+    it('should create a complete unit test config with all defaults', () => {
+      const config = createVitestConfig('test-package', 'unit');
+
+      expect(config).toEqual(
+        expect.objectContaining({
+          cacheDir: '../../node_modules/.vite/test-package',
+          test: expect.objectContaining({
+            reporters: ['basic'],
+            globals: true,
+            cache: { dir: '../../node_modules/.vitest' },
+            alias: expect.any(Array),
+            pool: 'threads',
+            poolOptions: { threads: { singleThread: true } },
+            environment: 'node',
+            include: [
+              'src/**/*.unit.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
+              'src/**/*.type.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
+            ],
+            globalSetup: ['../../global-setup.ts'],
+            setupFiles: expect.arrayContaining([
+              '../../testing/test-setup/src/lib/console.mock.ts',
+              '../../testing/test-setup/src/lib/reset.mocks.ts',
+              '../../testing/test-setup/src/lib/fs.mock.ts',
+            ]),
+            coverage: expect.objectContaining({
+              reporter: ['text', 'lcov'],
+              reportsDirectory: '../../coverage/test-package/unit-tests',
+              exclude: ['mocks/**', '**/types.ts', 'perf/**'],
+            }),
+            typecheck: { include: ['**/*.type.test.ts'] },
+          }),
+        }),
+      );
```

Comment:
No properties other than `cacheDir` and `test` should be created, so `expect.objectContaining` shouldn't be needed.